internal/controller/progressive/progressive.go

```diff
@@ -390,6 +384,54 @@ func AssessUpgradingPipelineType(ctx context.Context, existingUpgradingChildDef
 	return apiv1.AssessmentResultUnknown, nil
 }
 
+/*
+declareSuccess handles the success logic for the update of a child resource during a progressive upgrade.
+It patches both the existing and upgrading children, updates the child status and marks the rollout as deployed.
+
+Parameters:
+- ctx: The context for managing request-scoped values, cancellation, and timeouts.
+- rolloutObject: The current rollout object (this could be from cache).
```

Comment:
actually, let's not say "this could be from cache", since the PR I created today makes it so that it's not